tests/Unit/DataStructures/Tensor/Expressions/Test_Divide.cpp

```diff
@@ -0,0 +1,252 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <climits>
+#include <cmath>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace {
+// \brief Test the division of a tensor expression over a `double` is correctly
+// evaluated
+//
+// \details
+// The cases tested are:
+// - \f$L_{ij} = S_{ij} / R\f$
+// - \f$L_{ij} = (R * S_{ij} / T) / G\f$
+//
+// where \f$R\f$, \f$T\f$, and \f$G\f$ are `double`s and \f$S\f$ and \f$L\f$ are
+// Tensors with data type `double` or DataVector.
+//
+// \tparam DataType the type of data being stored in the numerator
+template <typename Generator, typename DataType>
+void test_divide_double(const gsl::not_null<Generator*> generator,
+                        const DataType& used_for_size) noexcept {
+  std::uniform_real_distribution<> distribution(-2.0, 2.0);
+  constexpr size_t dim = 3;
+
+  const auto S = make_with_random_values<tnsr::ii<DataType, dim>>(
+      generator, distribution, used_for_size);
+
+  // \f$L_{ij} = S_{ij} / R\f$
+  // Use explicit type (vs auto) for LHS Tensor so the compiler checks the
+  // return type of `evaluate`
+  const tnsr::ii<DataType, dim> Lij_from_Sij_over_R =
+      TensorExpressions::evaluate<ti_i, ti_j>(S(ti_i, ti_j) / 4.3);
+  // \f$L_{ij} = (R * S_{ij} / T / G)\f$
+  const tnsr::ii<DataType, dim> Lij_from_R_Sij_over_T =
+      TensorExpressions::evaluate<ti_i, ti_j>((-5.2 * S(ti_i, ti_j) / 1.6) /
+                                              2.1);
+
+  for (size_t i = 0; i < dim; i++) {
+    for (size_t j = 0; j < dim; j++) {
+      CHECK_ITERABLE_APPROX(Lij_from_Sij_over_R.get(i, j), S.get(i, j) / 4.3);
+      CHECK_ITERABLE_APPROX(Lij_from_R_Sij_over_T.get(i, j),
+                            -5.2 * S.get(i, j) / 1.6 / 2.1);
+    }
+  }
+}
+
+// \brief Test the division of a tensor expression over a rank 0 tensor
+// expression is correctly evaluated
+//
+// \details
+// The cases tested are:
+// - \f$L_{i}{}^{j} = S^{i}{}_{j} / R\f$
+// - \f$L^{k}{}_{i} = (R T_{i}{}^{k}) / (T_{j}{}^{l} S^{j}{}_{l})\f$
+// - \f$L_{i}{}^{k} = T_{i}{}^{k} / R^2 / R\f$
+//
+// where \f$R\f$, \f$S\f$, \f$T\f$, and \f$L\f$ are Tensors with data type
+// `double` or DataVector.
+//
+// \tparam DataType the type of data being stored in the operands and result
+template <typename Generator, typename DataType>
+void test_divide_rank0(const gsl::not_null<Generator*> generator,
+                       const DataType& used_for_size) noexcept {
+  std::uniform_real_distribution<> distribution(-2.0, 2.0);
```

Comment:
I think the issue with the failing test is that this distribution includes zero, but you use it to generate `R` which you divide by.